libraries/standard/mqtt/src/mqtt.c

```diff
@@ -433,18 +441,35 @@ static int32_t recvExact( const MQTTContext_t * pContext,
                                pIndex,
                                bytesRemaining );
 
-        if( bytesRecvd >= 0 )
+        if( bytesRecvd < 0 )
         {
-            bytesRemaining -= ( size_t ) bytesRecvd;
-            totalBytesRecvd += ( int32_t ) bytesRecvd;
-            pIndex += bytesRecvd;
+            LogError( ( "Network error while receiving packet: ReturnCode=%d.",
+                        bytesRecvd ) );
+            totalBytesRecvd = bytesRecvd;
+            receiveError = true;
         }
-        else
+        else if( bytesRecvd > ( int32_t ) bytesRemaining )
```

Comment:
How is it possible for the receive function to return more bytes than you ask for? The idea for this pattern is to limit the received bytes to fit into your buffer, basically thou shalt not receive more than this, so you should be guaranteed that this condition is never true.

Reply:
If there is a bug in the applications implementation.

Reply:
per offline conversation this will become an assert.